blockchain/treasury.go

```diff
@@ -880,8 +880,11 @@ type tspendVotes struct {
 	no    int    // No vote tally
 }
 
-// tSpendCountVotes returns the vote tally for a given tspend.
-// NOTE: This function must only be called on a TVI.
+// tSpendCountVotes returns the vote tally for a given tspend up to the
+// specified block. Note that this function errors if the block is outside the
+// voting window for the given tspend.
+//
+// This function must be called with the block index read lock held.
```

Comment:
This really shouldn't be needed for this func.  Looking at it more closely, I see that there is a call to `lookupNode` that is the culprit, but there is no reason to do that.  All that is needed is:

```Go
node = node.parent
```

Then there is no need to hold the lock.

Reply:
Good observation. Also changed in `sumPastTreasuryExpenditure` and reworked the commits.